src/CrossWord.java

```diff
@@ -1,102 +1,129 @@
+    int passwordChoice;
+
+    public CrossWord( int sizeRow, int sizeColumn ) {
+        array = new char[sizeRow][sizeColumn];
+        userArray = new char[sizeRow][sizeColumn];
     }
-    public void setCrossword()
-    {
-        array[0][1]='w';
-        array[1][1]='o';
-        array[2][1]='�';
-        array[3][1]='o';
-        array[4][1]='w';
-        array[5][1]='e';
 
-        array[1][0]='p';
-        array[1][2]='r';
+    public void setCrossword() {
+        array[0][1] = 'w';
+        array[1][1] = 'o';
+        array[2][1] = 'l';
+        array[3][1] = 'o';
+        array[4][1] = 'w';
+        array[5][1] = 'e';
 
-        array[3][0]='l';
-        array[3][2]='s';
+        array[1][0] = 'p';
+        array[1][2] = 'r';
 
-        for (int i=0;i<array.length;i++)
-        {
-            for(int j=0;j<array[i].length;j++)
-            {
-                if(array[i][j]==0)
-                {
-                    array[i][j]='_';
-                }
+        array[3][0] = 'l';
+        array[3][2] = 's';
+    }
+
+    public void playCrossword() {
+        do {
+            System.out.println("1: 2 Pionowo: mi�so na tatara");
+            System.out.println("2: 2 Poziomo: d�uga ro�lina warzywna");
+            System.out.println("3: 4 Poziomo: koleje ludzkiego �ycia");
+            System.out.println("Wpisz numer has�a: ");
+            passwordChoice = scanner.nextInt();
+            switch (passwordChoice) {
+                case 1:
+                    typeAnswer(0, 1);
+                    printCrossword(userArray);
+                    checkPassword(0, 1);
+                    break;
+
+                case 2:
+                    typeAnswer(1, 0);
+                    printCrossword(userArray);
+                    checkPassword(1, 0);
+                    break;
+                case 3:
+                    typeAnswer(3, 0);
+                    printCrossword(userArray);
+                    checkPassword(3, 0);
+                    break;
             }
         }
+        while (!checkCrossword());
     }
-    public void playCrossword()
-    {
-        System.out.println("1: 2 Pionowo: mi�so na tatara");
-        System.out.println("2: 2 Poziomo: d�uga ro�lina warzywna");
-        System.out.println("3: 4 Poziomo: koleje ludzkiego �ycia");
-        char type;
-        int choice=scanner.nextInt();
-        switch (choice)
-        {
-            case 1:
-                typeAnswer(0,1);
-                printCrossword(userArray);
-                checkCrossword(userArray);
-                break;
-            case 2:
 
+    private void typeAnswer( int numberRow, int numberColumn ) {
+        char type;
+        int i = numberRow;
+        int j = numberColumn;
+        if (passwordChoice == 1) {
+            for (i = 0; i < array.length; i++) {
+                type = scanner.next().charAt(0);
+                userArray[i][j] = type;
+            }
+        } else if (passwordChoice == 2 || passwordChoice == 3) {
```

Comment:
jeśli masz tylko opcje: 1, 2, 3 to niepotrzebne sprawdzenie czy 2 lub 3. Bo jeśli nie 1 to 2 lub 3.